dispatcher/dispatcher_test.go

```diff
@@ -244,50 +245,51 @@ func TestDo(t *testing.T) {
 }
 
 func TestValidate(t *testing.T) {
-	t.Run("return error if maxIter < 1", func(t *testing.T) {
-		const (
-			numWorker = 10
-			maxIter   = 0
-			expMsg    = "invalid value: maxIter: must be < 1, got 0"
-		)
-
-		err := dispatcher.New(numWorker).Do(context.Background(), maxIter, func() {})
-		checkErrorMessage(t, err, expMsg)
-	})
-
-	t.Run("return error if maxIter < numWorker", func(t *testing.T) {
-		const (
-			numWorker = 10
-			maxIter   = 5
-			expMsg    = "invalid value: maxIter: must be >= numWorker (10), got 5"
-		)
-
-		err := dispatcher.New(numWorker).Do(context.Background(), maxIter, func() {})
-		checkErrorMessage(t, err, expMsg)
-	})
-
-	t.Run("return error if callback == nil", func(t *testing.T) {
-		const (
-			numWorker = 10
-			maxIter   = 20
-			expMsg    = "invalid value: callback: must be non-nil"
-		)
-
-		err := dispatcher.New(numWorker).Do(context.Background(), maxIter, nil)
-		checkErrorMessage(t, err, expMsg)
-	})
-
-	t.Run("return nil if values are valid", func(t *testing.T) {
-		const (
-			numWorker = 10
-			maxIter   = 20
-		)
+	testcases := []struct {
+		label     string
+		exp       error
+		numWorker int
+		maxIter   int
+		callback  func()
+	}{
+		{
+			label:     "return error if maxIter < 1",
+			exp:       errors.New("invalid value: maxIter: must be < 1, got 0"),
```

Comment:
`exp: errors.New("invalid value: maxIter: must be >= 1, got 0")`
I think it should be **>=1**?